sdk/monitor/monitor-query/README.md

```diff
@@ -0,0 +1,174 @@
+[API reference documentation]<!--(https://docs.microsoft.com/javascript/api/@azure/monitor-query)--> |
+[Product documentation][azure_monitor_product_documentation]
+[Samples][samples]
+
+## Getting started
+
+### Install the package
+
+```bash
+npm install @azure/monitor-query
+```
+
+### Prerequisites
+
+- You must have an [Azure Subscription](https://azure.microsoft.com) and an [Azure Monitor][azure_monitor_product_documentation] resource to use this package.
+- Node.js version 8.x.x or higher
+
+### Create an Azure Monitor resource
+
+You can use the [Azure Portal][azure_monitor_create_using_portal] or the [Azure CLI][azure_monitor_create_using_cli] to create an Azure Monitor resource.
+
+Instructions:
+
+- [Using the Azure Portal][azure_monitor_create_using_portal]
+- [Using the Azure CLI][azure_monitor_create_using_cli]
+
+### Authenticate the client
+
+[LogsClient]<!--[msdocs_logs_client]--> and [MetricsClient]<!--[msdocs_metrics_client]--> authenticate using a [service principal](#authenticating-with-a-service-principal).
+
+#### Authenticating with a service principal
+
+Authentication via service principal is done by:
+
+- Creating a credential using the `@azure/identity` package.
+- Setting appropriate RBAC rules on your Azure Monitor resource.
+  More information on Azure Monitor roles can be found [here][azure_monitor_roles].
+
+Using [DefaultAzureCredential](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/identity/identity/README.md#defaultazurecredential)
+
+```javascript
+const { DefaultAzureCredential } = require("@azure/identity");
+const { LogsClient, MetricsClient } = require("@azure/monitor-query");
+
+const credential = new DefaultAzureCredential();
+
+const logsClient = new LogsClient(credential);
+// or
+const metricsClient = new MetricsClient(credential);
+```
+
+More information about `@azure/identity` can be found [here](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/identity/identity/README.md)
+
+## Key concepts
+
+The [`LogsClient`]<!--[msdocs_logs_client]--> allows you to query logs, using the [Kusto query language][kusto_query_language]. This data can be queried in the
+portal using tables like `AppEvents`, `AppDependencies` and others.
+
+The [`MetricsClient`]<!--[msdocs_metrics_client]--> allows you to query metrics.
+
+## Examples
+
+#### Querying logs
+
+The LogsClient can be used to query a Monitor workspace using the Kusto Query language.
+
+```javascript
+const monitorQuery = require("@azure/monitor-query");
+const { DefaultAzureCredential } = require("@azure/identity");
+
+const azureLogAnalyticsWorkspaceId = "<the Workspace Id for your Azure Log Analytics resource>";
+const logsClient = new monitorQuery.LogsClient(new DefaultAzureCredential());
+
+async function run() {
+  const kustoQuery = "AppEvents | limit 1";
+  const result = logsClient.queryLogs(azureLogAnalyticsWorkspaceId, kustoQuery);
+  const tablesFromResult: Table[] | undefined = result.tables;
```

Comment:
this is a js code snippet but we have a TS type annotation in it :)

Reply:
Fixed...:|